app/models/profile.rb

```diff
@@ -4,6 +4,9 @@ class Profile < ApplicationRecord
   validates :data, presence: true
   validates :user_id, uniqueness: true
   validates :location, :website_url, length: { maximum: 100 }
+  validates :website_url,
+    format: { with: URI::DEFAULT_PARSER.make_regexp(%w[https http]) },
+    if: Proc.new {|p| p.website_url.present? }
```

Comment:
```suggestion
   allow_nil: true
```

I tried this locally, all specs still pass.